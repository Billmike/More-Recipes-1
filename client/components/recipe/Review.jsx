import React from 'react';

const Review = () => (
  <li className="collection-item avatar">
    <img src="../../public/images/user.jpg" alt="" className="circle" />
    <span className="name"><b>John Stew</b></span>
    <p className="created-on">Oct 20, 2017 at 07:30:25 PM</p>
    <p id="review-content">
      A wonderful recipe, very tasty and easy to prepare.
    </p>
  </li>
);

export default Review;
