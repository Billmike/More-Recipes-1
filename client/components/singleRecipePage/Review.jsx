import React from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';

import userAvatar from '../../public/images/user_avatar_1.png';
import { decodeToken } from '../../utils/authenticate';

const propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number,
    recipeId: PropTypes.number,
    User: PropTypes.shape({
      fullName: PropTypes.string,
      userImage: PropTypes.string,
    }),
    content: PropTypes.string,
    createdAt: PropTypes.string,
  }).isRequired,
  handleDeleteReview: PropTypes.func.isRequired,
};

/**
 * Function to format date
 *
 * @param {string} date - The date to be formated
 *
 * @returns {string} - The formated date
 */
const createdOn = date => new Date(date).toLocaleString();

/**
 * Review component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const Review = (props) => {
  const { review, handleDeleteReview } = props;
  const {
    id, recipeId, userId, content, createdAt, User: { fullName, userImage }
  } = review;

  return (
    <li className="collection-item avatar">
      { review &&
      <div>
        <img src={userImage || userAvatar} alt="" className="circle" />
        <span className="name"><b>{fullName}</b></span>
        <p className="created-on">
          {createdOn(createdAt)}
          {userId === decodeToken().user.id &&
          <i
            role="button"
            tabIndex="0"
            className="material-icons delete-review right"
            onClick={() => handleDeleteReview(recipeId, id)}
            data-tip="Delete"
          >
            delete
          </i>
          }
        </p>
        <p className="review-content">
          {content}
        </p>
        <ReactTooltip />
      </div>
      }
    </li>
  );
};

Review.propTypes = propTypes;

export default Review;
