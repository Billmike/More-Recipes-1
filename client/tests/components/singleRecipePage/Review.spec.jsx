import React from 'react';

import Review from '../../../components/singleRecipePage/Review';

describe('Review component', () => {
  it('should render component', () => {
    const props = {
      review: {
        id: 2,
        recipeId: 3,
        userId: 1,
        User: {
          fullName: 'John Cane',
          userImage: 'User image url',
        },
        content: 'Wonderful recipe idea',
        createdAt: 'createdAt',
      },
      handleDeleteReview: jest.fn(),
    };
    const wrapper = shallow(<Review {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
