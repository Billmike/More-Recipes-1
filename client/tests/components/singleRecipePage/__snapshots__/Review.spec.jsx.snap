// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review component should render component 1`] = `
<li
  className="collection-item avatar"
>
  <div>
    <img
      alt=""
      className="circle"
      src="User image url"
    />
    <span
      className="name"
    >
      <b>
        John Cane
      </b>
    </span>
    <p
      className="created-on"
    >
      Invalid Date
    </p>
    <p
      className="review-content"
    >
      Wonderful recipe idea
    </p>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </div>
</li>
`;
